mtp_transaction_uploader/upload.py

```diff
@@ -0,0 +1,125 @@
+import os
+import shutil
+import re
+from datetime import datetime
+from collections import namedtuple
+
+from bankline_parser.data_services import parse_file
+from bankline_parser.data_services.enums import TransactionCode
+from pysftp import Connection
+
+from .api_client import get_authenticated_connection
+from . import settings
+
+ref_pattern = re.compile('([A-Za-z][0-9]{4}[A-Za-z]{2}).*?([0-9]{1,2}).*?([0-9]{1,2}).*?([0-9]{2,4})')
+file_pattern = re.compile('YO1A\.REC\.#D\.(%(code)s)\.D([0-9]{6})' % {'code': settings.ACCOUNT_CODE})
+
+
+def retrieve_data_services_files():
+    # check for existing downloaded files and remove if found
+    if os.path.exists(settings.DS_NEW_FILES_DIR):
+        shutil.rmtree(settings.DS_NEW_FILES_DIR)
+    os.mkdir(settings.DS_NEW_FILES_DIR)
+
+    # check if we recorded date of last file downloaded
+    last_date = None
+    if (os.path.exists(settings.DS_LAST_DATE_FILE)):
+        with open(settings.DS_LAST_DATE_FILE) as f:
+            last_date_str = f.read()
+            last_date = datetime.strptime(last_date_str, '%d%m%y')
+
+    new_files = []
+    new_dates = []
+    with Connection(settings.SFTP_HOST, username=settings.SFTP_USER,
+                    private_key=settings.SFTP_PRIVATE_KEY) as conn:
+        with conn.cd(settings.SFTP_DIR):
+            dir_listing = conn.listdir()
+            for filename in dir_listing:
+                m = file_pattern.match(filename)
+
+                if m:
+                    date = datetime.strptime(m.group(2), '%d%m%y')
+                    if date > last_date:
+                        local_path = os.path.join(settings.DS_NEW_FILES_DIR,
+                                                  filename)
+                        new_files.append(local_path)
+                        new_dates.append(date)
+                        conn.get(filename, local_path=local_path)
+
+    # find last dated file
+    if len(new_dates) > 0:
+        new_last_date = sorted(new_dates)[-1]
+
+    FileDownload = namedtuple('FileDownload', ['new_last_date', 'new_files'])
+    return FileDownload(new_last_date, new_files)
+
+
+def upload_transactions_from_files(files):
+    transactions = []
+    for filename in files:
+        transactions += get_transactions_from_file(filename)
+    conn = get_authenticated_connection()
+    conn.bank_admin.transactions.post(transactions)
```

Comment:
Will depend on web server configuration - both the actual option that allows you to configure the max request body size, and the hosts performance characteristics.

Would be good to stress test on a production-like environment.